R/pkg/R/DataFrame.R

```diff
@@ -1384,7 +1384,7 @@ setMethod("saveAsTable",
                                     "org.apache.spark.sql.parquet")
             }
             allModes <- c("append", "overwrite", "error", "ignore")
-            if (!(mode %in% allModes)) {
+            if (! (mode %in% allModes)) {
```

Comment:
I'm with @sun-rui. I prefer the following style: 
```!<var>, !<function call> and !(<expression>)```

Unfortunately, we can't currently customize the linter about a space before left parenthesis to do that. So I think we should add a new rule into lintr in itself. Otherwise, should we turn off the linter?